Source/WebCore/dom/InternalObserverInspect.cpp

```diff
@@ -0,0 +1,252 @@
+                m_subscriber->error(exception->value());
+                return;
+            }
+        }
+
+        m_subscriber->next(value);
+    }
+
+    void error(JSC::JSValue value) final
+    {
+        removeAbortHandler();
+
+        if (m_inspector.error) {
+            Ref vm = protectedGlobalObjectVM();
+            JSC::JSLockHolder lock(vm);
+            auto scope = DECLARE_CATCH_SCOPE(vm);
+
+            m_inspector.error->handleEventRethrowingException(value);
+
+            JSC::Exception* exception = scope.exception();
+            if (UNLIKELY(exception)) {
+                scope.clearException();
+                m_subscriber->error(exception->value());
+                return;
+            }
+        }
+
+        m_subscriber->error(value);
+    }
+
+    void complete() final
+    {
+        InternalObserver::complete();
+
+        removeAbortHandler();
+
+        if (m_inspector.complete) {
+            Ref vm = protectedGlobalObjectVM();
+            JSC::JSLockHolder lock(vm);
+            auto scope = DECLARE_CATCH_SCOPE(vm);
+
+            m_inspector.complete->handleEventRethrowingException();
+
+            JSC::Exception* exception = scope.exception();
+            if (UNLIKELY(exception)) {
+                scope.clearException();
+                m_subscriber->error(exception->value());
+                return;
+            }
+        }
+
+        m_subscriber->complete();
+    }
+
+    void visitAdditionalChildren(JSC::AbstractSlotVisitor& visitor) const final
+    {
+        m_subscriber->visitAdditionalChildren(visitor);
+        if (m_inspector.next)
+            m_inspector.next->visitJSFunction(visitor);
+        if (m_inspector.error)
+            m_inspector.error->visitJSFunction(visitor);
+        if (m_inspector.complete)
+            m_inspector.complete->visitJSFunction(visitor);
+        if (m_inspector.subscribe)
+            m_inspector.subscribe->visitJSFunction(visitor);
+        if (m_inspector.abort)
+            m_inspector.abort->visitJSFunction(visitor);
+    }
+
+    void visitAdditionalChildren(JSC::SlotVisitor& visitor) const final
+    {
+        m_subscriber->visitAdditionalChildren(visitor);
+        if (m_inspector.next)
+            m_inspector.next->visitJSFunction(visitor);
+        if (m_inspector.error)
+            m_inspector.error->visitJSFunction(visitor);
+        if (m_inspector.complete)
+            m_inspector.complete->visitJSFunction(visitor);
+        if (m_inspector.subscribe)
+            m_inspector.subscribe->visitJSFunction(visitor);
+        if (m_inspector.abort)
+            m_inspector.abort->visitJSFunction(visitor);
+    }
+
+    void removeAbortHandler()
+    {
+        if (!m_abortAlgorithmHandler)
+            return;
+
+        auto handle = std::exchange(m_abortAlgorithmHandler, std::nullopt);
+        m_subscriber->protectedSignal()->removeAlgorithm(*handle);
+    }
+
+    Ref<JSC::VM> protectedGlobalObjectVM() const
```

Comment:
This is only used to initialize local variables, it doesn't need to return a smart pointer and doesn't need to have `protected` in the name. I would just use:
`JSC::VM* vm() const;`

Then update the call sites to call `vm()` instead of `protectedGlobalObjectVM()`